### DISCUSSION TITLE
allow watching of the node_modules folder

### DISCUSSION THREAD

**Discussion by chris-kruining** [remix-run/remix#2626] [Proposals]
I am developing a local components library which I want to use in my remix project.

However every time I make a change and recompile my lib I have to kill remix and restart it in order for it to pick up the changes to the lib.

I would therefor like to have an option to also watch for changes in the node_modules folder.

**Comment by kiliman** [comment]
You can use `nodemon` to watch the `node_modules` folder and restart.

If you're using Express, simply replace the call to `node server.js` with `nodemon --watch node_modules server.js`

If you're using Remix App Server, you'll need to explicitly call the remix cli script
```bash
nodemon --watch node_modules node_modules/@remix-run/dev/cli.js dev
```

**Reply by chris-kruining** [reply]
how would this translate when using `cross-env` in the script?:

```json
"dev": "cross-env PORT=3000 REMIX_DEV_SERVER_WS_PORT=8002 NODE_ENV=development remix dev",
```

**Reply by chris-kruining** [reply]
I got it running with 

```json
"dev": "cross-env PORT=3000 REMIX_DEV_SERVER_WS_PORT=8002 NODE_ENV=development nodemon --watch ./node_modules --exec ./node_modules/.bin/remix dev",
```

but it is not picking up my changes though

**Reply by kiliman** [reply]
So is nodemon restarting when you change your file?

**Reply by chris-kruining** [reply]
no it isn't

**Reply by kiliman** [reply]
You might be better off switching to the Express adapter and managing your own server instead of the built-in one.

**Comment by Esarhaddon** [comment]
For anyone else in a similar situation that comes across this while googling, the `watchPaths` option in `remix.config.js` solves this problem nicely for me, e.g. `watchPaths: ["./node-modules/my-dev-package"]`.
